### ISSUE TITLE
Security: Bad IP connecting

### ISSUE THREAD

**Issue by jjay225** [hgupta9/FluentFTP#121]
Hi there,

Cool package, thanx :) I'm having an issue connecting to a Vsftpd unix ftps server. Whenever I do a "GetListing()" I get the error message: Security: Bad IP connecting. Now from my research this can be fixed by added a config line to the ftp server but I'm not under control of this server, have you had this issue before and know of a workaround? Check the below links for reference:
[https://forum.ivorde.com/vsftpd-425-security-bad-ip-connecting-error-message-t100.html](url)
[http://www.linuxquestions.org/questions/linux-newbie-8/vsftpd-problem-with-425-security-bad-ip-connecting-120158/](url)

Thanx again

**Comment by hgupta9** [comment]
GetListing() starts another FTP connection to the server to transfer the data.

You can try passive mode.

- `client.DataConnectionType = FtpDataConnectionType.PASV`

You can also try PASVEX passive mode which .

- `client.DataConnectionType = FtpDataConnectionType.PASVEX`

And please use the beta version which improves logging quality:
- [FluentFTP_17.4.3_BETA4.zip](https://we.tl/T8c3mkLZLq)

If this still does not work please try connecting with [FileZilla](https://filezilla-project.org/), if it works, then copy the entire FileZilla log as well as the entire [FluentFTP log](https://github.com/hgupta9/FluentFTP#faq_trace), so I can compare both and see what FileZilla got right.

**Comment by jjay225** [comment]
Thanks for your help :) I see the problem ended up being a network issue more so. I was able to connect and download a file so that's all I need. I will contact you again if there's an issue. The issue is: I was connecting from my pc in a office on a LAN. I then copied my utility console test app to the production server and it works from there. 

I think what happens is I connect from my machine, the remote ftp server sees the company external IP, all is fine. Then I try list files and the server now sees another different IP, being my internal IP I think it will disconnect as it's configured to not allow to simultaneous IP's.
